src/app/plugins/pubsub.plugin.ts

```diff
@@ -1,23 +1,142 @@
+  private _VALID_KEYS: string[] = ['random', 'list', 'types', ''];
+  private _DEFAULT_INPUT: string = 'chicken-tenders';
+  private _SUBS: string[] = [];
+  private _EMBED_LIST = new MessageEmbed();
+
+  private _SUB_UPD_THRESH: number = 1000 * 60 * 60 * 24; // in ms, one day.
+  private _LAST_UPD_TIME: number = 0;
 
   constructor(public container: IContainer) {
     super();
+    this._updateList();
+  }
+
+  private async _updateList() {
+    this.container.httpService
+      .get(`${this._API_URL}/allsubs/`)
+      .then((response: IHttpResponse) => {
+        const subs = response.data;
+
+        this._SUBS = subs.map((subData: { name: string }) => subData.name);
+      })
+      .catch((err) => this.container.loggerService.warn(err));
   }
 
   public async execute(message: IMessage, args?: string[]) {
-    const state = this.container.storeService.get('pubSubStore').state;
-    if (state?.onSale) {
-      message.reply('Chicken tender subs are on sale :eyes:');
-    } else {
-      message.reply('Chicken tender subs are **NOT** on sale.');
+    const input = (args || []).join('-').toLowerCase() || this._DEFAULT_INPUT;
+
+    if (input === 'list' || input === 'types') {
+      //Simply return the list of supported subs
+      await this._generateEmbedList();
+      message.reply(this._EMBED_LIST);
+      return;
+    }
+
+    // checks if their sub was a sub, then if that sub is recognised
+    const subType = this._SUBS.find((sub: string) => sub === input) || 'random';
+
+    //recieves the according info and posts, or derps
+    await this.container.httpService
+      .get(`${this._API_URL}/subs/?name=${subType}`)
+      .then((response: IHttpResponse) => {
+        const [subData] = response.data;
+        const embed: MessageEmbed = this._generateEmbedSub(subData);
+
+        message.reply(embed);
+      })
+      .catch((err) => this.container.loggerService.warn(err));
+  }
+
+  private async _generateEmbedList(): Promise<void> {
+    const lastListUpdate: number = Date.now() - this._LAST_UPD_TIME;
+
+    if (lastListUpdate < this._SUB_UPD_THRESH) {
+      return;
+    }
+
+    this._LAST_UPD_TIME = Date.now(); //since it only updates once a day, don't need to worry about accuracy
+
+    await this._updateList();
+
+    //setting up number of columns and rows for the list
+    const maxCol = 3;
+    const maxRows = 10;
+    const numCols = Math.min(maxCol, Math.ceil(this._SUBS.length / maxRows));
+    const numRows = Math.ceil(this._SUBS.length / numCols);
+
+    //sets up a new embed for the list to reside in
+    this._EMBED_LIST = new MessageEmbed();
+    this._EMBED_LIST.setColor('#0099ff').setTitle('Types of Subs');
+    const subs = this._SUBS.map((sub) => this._normalizeName(sub));
+
+    //cycles through each column inserting their row, and also notes the alphabetic range
+    for (let cols = 0; cols < numCols; cols++) {
+      const columnSubs = subs.slice(numRows * cols, numRows * (cols + 1));
```

Comment:
nit: i hate `for` loops. i dont like this pattern. is there more functional way to do?